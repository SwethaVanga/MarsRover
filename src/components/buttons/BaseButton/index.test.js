import React from "react";
import { shallow } from 'enzyme';
import renderer from "react-test-renderer";
import ButtonBase from "./index";

test("Should render button component with props", () => {
    const component = renderer.create(<ButtonBase clickAction={()=> {}} title={"Run"} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test("Test click event", () => {
    const mockCallBack = jest.fn();
    const button = shallow(<ButtonBase clickAction={mockCallBack} title={"Hide"} />);
    button.find("button").simulate("click");
    expect(mockCallBack.mock.calls.length).toEqual(1);
});
