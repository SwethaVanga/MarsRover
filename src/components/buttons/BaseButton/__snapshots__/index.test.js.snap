// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render button component with props 1`] = `
<div
  className="header"
>
  <div
    className="button white"
  >
    <p>
      <span
        className="bg"
      />
      <span
        className="base"
      />
      <span
        className="text"
        onClick={[Function]}
      >
        Run
      </span>
    </p>
  </div>
</div>
`;
